bf2fc6cc711aee1a0c2a/kas-fleet-manager
# Automate the configuration of OpenID Identity Provider for each new Data Plane OSD cluster

## Description


Resolves https://issues.redhat.com/browse/MGDSTRM-2026 and https://issues.redhat.com/browse/MGDSTRM-2027

**The IDP name should not be changed.** 

## TODO:
- [x] Update Vault with the new secrets
- [x] ~~Update AppInterface to take into account the new vault secret and the new realm.~~ See https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/15994

/cc @davidffrench @david-martin 

## Verification Steps

**Note** - You'll need an account in ma-sso-playground to do the verification to the very end. Have a chat with @akoserwal to create one for you. 

1. Make sure you have your own OSD cluster (either existing or a new one) 
2. Seed the details of that cluster in the database using `make db/generate/insert/cluster CLUSTER_ID=<cluster-id>`
3. Run the API using `make run`
4. Observes the logs to make sure that the cluster_mgr has run. The log that should interest you the most is  `reconciled ready cluster .....`
5. Go to details of your cluster is OCM UI, and click on the `AccessControl` tab. You should see a new configured `MAS-SSO` IDP.
6. Go the login page of OSD console, observe the new `MAS-SSO` IDP listed among the login options. 
7. Click on it and use your MAS-SSO credentials to login in the cluster.  

## Type of change


## Checklist (Definition of Done)
